services/src/main/scala/cromwell/pipeline/service/UserService.scala

```diff
@@ -11,7 +11,7 @@ import scala.util.Random
 
 trait UserService {
 
-  def getUsersByEmail(emailPattern: String): Future[Seq[User]]
+  def getUsersByEmail(emailPattern: String): Future[Seq[UserNoCredentials]]
 
   def getUserByEmail(email: UserEmail): Future[Option[User]]
```

Comment:
Shouldn't the same approach be implemented to that method as well?

Reply:
We need this method for authentication, so it must return a full user. However, I suggest renaming it to `getUserWithCredsByEmail` or something like that to explicitly show that it shouldn't be used arbitrarily.